server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregatorTestBase.java

```diff
@@ -0,0 +1,147 @@
+        });
+    }
+
+    public void testFieldMissing() throws IOException {
+        testCase(new MatchAllDocsQuery(), "wrong_field", minPrecision, iw -> {
+            iw.addDocument(Collections.singleton(new LatLonDocValuesField(FIELD_NAME, 10D, 10D)));
+        }, geoGrid -> {
+            assertEquals(0, geoGrid.getBuckets().size());
+        });
+    }
+
+    public void testWithSeveralDocs() throws IOException {
+        int precision = randomPrecision();
+        int numPoints = randomIntBetween(8, 128);
+        Map<String, Integer> expectedCountPerGeoHash = new HashMap<>();
+        testCase(new MatchAllDocsQuery(), FIELD_NAME, precision, iw -> {
+            List<LatLonDocValuesField> points = new ArrayList<>();
+            Set<String> distinctHashesPerDoc = new HashSet<>();
+            for (int pointId = 0; pointId < numPoints; pointId++) {
+                double lat = (180d * randomDouble()) - 90d;
+                double lng = (360d * randomDouble()) - 180d;
+
+                // Precision-adjust longitude/latitude to avoid wrong bucket placement
+                lng = adjustLongitude(lng);
+                lat = adjustLatitude(lat);
+
+                points.add(new LatLonDocValuesField(FIELD_NAME, lat, lng));
+                String hash = hashAsString(lng, lat, precision);
+                if (distinctHashesPerDoc.contains(hash) == false) {
+                    expectedCountPerGeoHash.put(hash, expectedCountPerGeoHash.getOrDefault(hash, 0) + 1);
+                }
+                distinctHashesPerDoc.add(hash);
+                if (usually()) {
+                    iw.addDocument(points);
+                    points.clear();
+                    distinctHashesPerDoc.clear();
+                }
+            }
+            if (points.size() != 0) {
+                iw.addDocument(points);
+            }
+        }, geoHashGrid -> {
+            assertEquals(expectedCountPerGeoHash.size(), geoHashGrid.getBuckets().size());
+            for (GeoGrid.Bucket bucket : geoHashGrid.getBuckets()) {
+                assertEquals((long) expectedCountPerGeoHash.get(bucket.getKeyAsString()), bucket.getDocCount());
+            }
+            assertTrue(AggregationInspectionHelper.hasValue(geoHashGrid));
+        });
+    }
+
+    private void testCase(Query query, String field, int precision, CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
+                          Consumer<InternalGeoGrid<T>> verify) throws IOException {
+        Directory directory = newDirectory();
+        RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
+        buildIndex.accept(indexWriter);
+        indexWriter.close();
+
+        IndexReader indexReader = DirectoryReader.open(directory);
+        IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+
+        GeoGridAggregationBuilder aggregationBuilder = new GeoHashGridAggregationBuilder("_name").field(field);
```

Comment:
I think I would prefer to move things like this into GeoHashGridTests in this PR as well. It might give us a better idea about future separation of generic and specific pieces.